File: scripts/convert-favicon.py
#!/usr/bin/env python3
"""
Convert logo.png to favicon.ico and apple-touch-icon.png
"""
import sys
import os

try:
    from PIL import Image
except ImportError:
    print("Installing Pillow...")
    os.system(f"{sys.executable} -m pip install Pillow")
    from PIL import Image

def convert_to_favicon():
    # Paths
    logo_path = "public/logo.png"
    favicon_path = "public/favicon.ico"
    apple_touch_icon_path = "public/apple-touch-icon.png"
    
    # Open the logo
    img = Image.open(logo_path)
    
    # Convert to RGB if necessary (ICO doesn't support RGBA well)
    if img.mode == 'RGBA':
        # Create a white background
        background = Image.new('RGB', img.size, (255, 255, 255))
        background.paste(img, mask=img.split()[3])  # Use alpha channel as mask
        img = background
    elif img.mode != 'RGB':
        img = img.convert('RGB')
    
    # Create favicon.ico with multiple sizes
    favicon_sizes = [(16, 16), (32, 32), (48, 48)]
    favicon_images = []
    for size in favicon_sizes:
        resized = img.copy()
        resized.thumbnail(size, Image.Resampling.LANCZOS)
        favicon_images.append(resized)
    
    # Save as ICO
    favicon_images[0].save(
        favicon_path,
        format='ICO',
        sizes=favicon_sizes
    )
    print(f"✓ Created {favicon_path}")
    
    # Create apple-touch-icon (180x180)
    apple_icon = img.copy()
    apple_icon.thumbnail((180, 180), Image.Resampling.LANCZOS)
    apple_icon.save(apple_touch_icon_path, format='PNG')
    print(f"✓ Created {apple_touch_icon_path}")
    
    print("\n✓ Favicon conversion complete!")

if __name__ == "__main__":
    convert_to_favicon()

File: scripts/convert-favicon.js
const fs = require('fs');
const path = require('path');

async function convertToFavicon() {
  try {
    const sharp = require('sharp');
    
    const logoPath = path.join(__dirname, '../public/logo.png');
    const faviconPath = path.join(__dirname, '../public/favicon.ico');
    const appleTouchIconPath = path.join(__dirname, '../public/apple-touch-icon.png');
    
    // Read the logo
    const logoBuffer = fs.readFileSync(logoPath);
    
    // Create favicon.ico (64x64 for better visibility)
    await sharp(logoBuffer)
      .resize(64, 64, {
        fit: 'contain',
        background: { r: 255, g: 255, b: 255, alpha: 1 }
      })
      .toFormat('png')
      .toFile(faviconPath.replace('.ico', '-64.png'));
    
    // Rename to .ico
    fs.renameSync(faviconPath.replace('.ico', '-64.png'), faviconPath);
    console.log('✓ Created favicon.ico (64x64)');
    
    // Create apple-touch-icon (180x180)
    await sharp(logoBuffer)
      .resize(180, 180, {
        fit: 'contain',
        background: { r: 255, g: 255, b: 255, alpha: 1 }
      })
      .toFormat('png')
      .toFile(appleTouchIconPath);
    console.log('✓ Created apple-touch-icon.png');
    
    console.log('\n✓ Favicon conversion complete!');
  } catch (error) {
    if (error.code === 'MODULE_NOT_FOUND') {
      console.log('Installing sharp package...');
      const { execSync } = require('child_process');
      execSync('npm install sharp', { stdio: 'inherit' });
      console.log('\nPlease run this script again.');
    } else {
      console.error('Error:', error.message);
    }
  }
}

convertToFavicon();
